### PR TITLE
Force dns resolution on connection retry

### PR DESCRIPTION
**Pull request by eabbott**
DNS resolution is only done once on initialization. We can trigger memcached clients to reload through zookeeper. It appears many java systems do not listen for changed zookeeper params as evidenced after the past few memcached swaps. Forcing DNS resolution should provide a nice catch all solution for the stragglers.

With this in place, we'll be able to change the zookeeper param which will trigger many java apps to properly fail over to the new memcached server(s). The next step is always to flush the cache on the old memcached servers. Instead of flushing the cache via the memcached protocol, we can restart the process. The changes in this PR will be triggered by that connection being dropped and will reconnect to the new memcached servers.

### CODE REVIEW

**Comment by jaredstehler** [comment]
what kind of performance difference would you expect with this change?

**Comment by eabbott** [comment]
I would not expect to see a performance difference. The extra DNS resolution is done on reconnect, not on initial connect. On a reconnect, it was already not connected, so adding a small amount of time to the reconnection time doesn't seem a deal breaker. Also, it only runs on reconnect. Reconnect is done in the background on a exponential backoff alg with a max retry interval of 30s (that means the longest interval will be 30s between retries).

File: src/main/java/net/spy/memcached/protocol/TCPMemcachedNodeImpl.java [hunk 1]
```diff
@@ -364,6 +365,22 @@ public final SocketAddress getSocketAddress() {
 		return socketAddress;
 	}
 
+	private final void setSocketAddress(SocketAddress sa) {
```
Comment (jhaber) [inline src/main/java/net/spy/memcached/protocol/TCPMemcachedNodeImpl.java]: private final is redundant

**Comment by jhaber** [comment]
lgtm

**Comment by eabbott** [comment]
Removed redundant final keyword from private method.

**Pull request merged**